### ISSUE TITLE
Training not converge

### ISSUE THREAD

**Issue by yq1011** [ZheC/Realtime_Multi-Person_Pose_Estimation#19]
Hi,

Can you share how long does it take for you to train on what GPU? And what should the final loss be?

After 2 days' training on K80,  it just trained for 17100 iters and the loss is still in the range of 500-1000. Is this right?

Thanks

**Comment by ZheC** [comment]
With the large training dataset, the model convergence is slow. We use two Titanx (old version) and train for six days. You can change the batch size from 10 to 8 to speed it up a little bit. We did not try a batch size below 8.  I will plot the loss with respect to iterations and post here.

**Comment by yq1011** [comment]
Thanks a lot!

**Comment by ZheC** [comment]
![image](https://cloud.githubusercontent.com/assets/8867900/23180538/c04ba1be-f83f-11e6-82ad-54cfd09cfcdc.png)
The light blue line is training loss at stage 6, without any smoothing. The other ones show the loss at each stage, smoothed over around 200 samples (so if you're displaying at 5 iters, that's smoothed over 200*5 iterations).

**Comment by yq1011** [comment]
Hi, is this the loss of L1 or L2?

**Comment by yq1011** [comment]
and you are using a base_lr of 0.00002 here. In the solver generated in your code, it's 0.00004. Is 2e-5 better?

**Comment by yq1011** [comment]
hi, any updates? :D

**Comment by Shaswat27** [comment]
@ZheC Can you post the loss vs iterations curve again?

**Comment by sunguanxiong** [comment]
@yq1011 what's your final result? Was it converge? How long did you train for it? I am training this model for days, and it's not so fast as ZheC said, I use 4 GPUs trained for 2 days , and it's about 2w iterations...

**Comment by ds2268** [comment]
I think we should talk in terms of epochs instead (the training is printing that). @ZheC when you mentioned that you were using 2 x Titan X with batch size of 10 that probably meant that the actual batch size was 20 (10 per gpu) ?

**Comment by wujiyoung** [comment]
I have the same problem as you @yq1011 .
Did you get a converged loss finally?

**Comment by jricheimer** [comment]
@ZheC Hi, would you be able to post the loss curve again? I would like to compare with the model when I train it locally to make sure it is performing comparably. Thanks!

**Comment by ildoonet** [comment]
@ZheC Can you post the loss curve?

**Comment by ZheC** [comment]
Hi all, I am really sorry for my late response. I graduated from CMU so it is not easy to access the old files again. But I plot the loss for the two levels here:

https://github.com/ZheC/Realtime_Multi-Person_Pose_Estimation/blob/master/training/example_loss/Loss_l1.png

https://github.com/ZheC/Realtime_Multi-Person_Pose_Estimation/blob/master/training/example_loss/Loss_l2.png

All the terminal output is: https://github.com/ZheC/Realtime_Multi-Person_Pose_Estimation/blob/master/training/example_loss/output.txt

The code to plot the loss is here: https://github.com/ZheC/Realtime_Multi-Person_Pose_Estimation/blob/master/training/example_loss/plotLoss.sh

**Comment by ildoonet** [comment]
@ZheC Thanks for sharing. But I think the output is clipped.

```bash
KAKAOui-MacBook-Pro-2:~ ildoonet$ tail -10 ~/Downloads/output.txt
I0906 12:55:55.459972 23236 solver.cpp:244]     Train net output #3: loss_stage2_L2 = 27.5409 (* 1 = 27.5409 loss)
I0906 12:55:55.459980 23236 solver.cpp:244]     Train net output #4: loss_stage3_L1 = 93.887 (* 1 = 93.887 loss)
I0906 12:55:55.459985 23236 solver.cpp:244]     Train net output #5: loss_stage3_L2 = 26.4227 (* 1 = 26.4227 loss)
I0906 12:55:55.459991 23236 solver.cpp:244]     Train net output #6: loss_stage4_L1 = 92.5074 (* 1 = 92.5074 loss)
I0906 12:55:55.460000 23236 solver.cpp:244]     Train net output #7: loss_stage4_L2 = 25.912 (* 1 = 25.912 loss)
I0906 12:55:55.460005 23236 solver.cpp:244]     Train net output #8: loss_stage5_L1 = 91.5887 (* 1 = 91.5887 loss)
I0906 12:55:55.460011 23236 solver.cpp:244]     Train net output #9: loss_stage5_L2 = 25.4472 (* 1 = 25.4472 loss)
I0906 12:55:55.460016 23236 solver.cpp:244]     Train net output #10: loss_stage6_L1 = 90.5205 (* 1 = 90.5205 loss)
I0906 12:55:55.460022 23236 solver.cpp:244]     Train net output #11: loss_stage6_L2 = 24.9858 (* 1 = 24.9858 loss)
I0906 12:55:56.433954 23236 sgd_solver.cpp:106] Iteration 216345, lr = 6.66e-06
```

**Comment by ildoonet** [comment]
@ZheC Thanks for sharing!!

**Comment by Nestarneal** [comment]
@ZheC Hi, I set the parameters based on your terminal output and train, but it is terminated about iteration 1200 with no any log shown on the screen. Do you have any idea about this? Thanks.

**Comment by YongboYu** [comment]
Hi,@ZheC,thanks for your great work! I have some questions about train the pose model. In your https://github.com/ZheC/Realtime_Multi-Person_Pose_Estimation/blob/master/training/example_loss/Loss_l1.png

https://github.com/ZheC/Realtime_Multi-Person_Pose_Estimation/blob/master/training/example_loss/Loss_l2.png
,But there was only the result about iterations of # 25,0000. However, in the Openpose
https://github.com/CMU-Perceptual-Computing-Lab/openpose
the https://github.com/CMU-Perceptual-Computing-Lab/openpose/tree/master/models/pose/coco
the model was shown is pose_iter_# 440000.caffemodel, I would like to know whether it is only 250000 iterations or 440000 iterations.
Thanks for you attentions.

**Comment by YongboYu** [comment]
Are there someone getting the same results in the paper , and trying to train the small model only 2-stages used?

**Comment by YongboYu** [comment]
@yq1011 Did you get the same results as the paper?Thanks

**Comment by ZheC** [comment]
@Ai-is-light I use 440000 iterations' model. I pick up the best iteration based on the evaluation score on a validation set. I keep testing the accuracy of the trained models at different iteration.  The best iteration is not fixed for different models. So I think probably you want to follow the same way to pick up your trained model.

**Comment by yw155** [comment]
Hi @ZheC , I have a question that how much the effect the number of images being used in the evaluation have to the evaluation score. I saw on your paper you chosen 1000 images randomly. Why not choose the whole validation set? In addition, at the line 13 of the file 'evalCOCO.m', the variable 'coco_val' seems no explicitly definition. So I didn't know the length of it.
Thanks in advance.

**Comment by soans1994** [comment]
Hello,
How to choose the stepsize for a given iteration?

The default parametrs from the author is step size=13106 for iterations=600000

Thank You
